### PR TITLE
Better Support for using MKL as the Blas/Lapack Library

### PR DESCRIPTION
**Pull request by kylemacfarlan**
Thanks for maintaining this! It has made my life a lot easier. However, it would be nice to have better built-in support for building suitesparse with Intel's MKL. 

This PR adds a new option: USE_MKL. When this is ON (Off by default), it will use the FindMKL.cmake module to find an installed version of MKL. 

My cmake skills are not the best, so any suggestions for improvement are appreciated. 

Thanks,
Kyle

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,12 +6,14 @@
...
 
 option(HUNTER_ENABLED "Enable Hunter package manager support" OFF)
-include("cmake/HunterGate.cmake")
-HunterGate(
-  URL "https://github.com/ruslo/hunter/archive/v0.23.34.tar.gz"
-  SHA1 "70287b1ffa810ee4e952052a9adff9b4856d0d54"
-)
-
+option(USE_MKL "Use Intel MKL for blas/lapack" OFF)
+if (HUNTER_ENABLED)
```
Comment (NeroBurner) [inline CMakeLists.txt]: don't remove the HUNTER_ENABLED option, HunterGate.cmake itself handles the HUNTER_ENABLED option. So no need for the new extra `if()` block

Add just your new option (change of one line)

File: CMakeLists.txt [hunk 2]
```diff
@@ -216,71 +217,82 @@ if(WITH_CUDA)
...
-								${lapack_blas_windows_dll}
-					DESTINATION ${SuiteSparse_LAPACK_BLAS_LIB_DIR}
-			)
-		endif()
-	ENDIF()
-ENDIF()
-ENDIF() # LAPACK found
-endif() # HUNTER_ENABLED
 
+if (USE_MKL)
```
Comment (NeroBurner) [inline CMakeLists.txt]: big changes are hard to review, please leave the if-else block and add the new flag with an elif
```
if (HUNTER_ENABLED)
  original code
elseif (USE_MKL)
  your code
else()
  original block with BLAS and LAPACK
endif()

File: README.md [hunk 3]
```diff
@@ -6,6 +6,7 @@ CMake scripts for painless usage of Tim Davis' [SuiteSparse](http://www.cise.ufl
 
 The goal is using one single CMake code to build against *SuiteSparse* in standard Linux package systems (e.g. `libsuitesparse-dev`) and in manual compilations under Windows. 
 
+
```
Comment (NeroBurner) [inline README.md]: remove this change or add meaningful documentation

File: cmake/Findmkl.cmake [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                 NAMES ${INT_LIB}
+                 PATHS $ENV{MKLROOT}/lib
+                       $ENV{MKLROOT}/lib/ia32
+                       $ENV{INTEL}/mkl/lib/ia32
+                 NO_DEFAULT_PATH)
+endif()
+set(MKL_INCLUDE_DIRS ${MKL_INCLUDE_DIR})
+set(MKL_LIBRARIES ${MKL_SEQUENTIAL_LAYER_LIBRARY} ${MKL_INTERFACE_LIBRARY} ${MKL_CORE_LIBRARY})
+
+if (MKL_INCLUDE_DIR AND
```
Comment (NeroBurner) [inline cmake/Findmkl.cmake]: this should already be handled by FIND_PACKAGE_HANDLE_STANDARD_ARGS, please remove

File: cmake/Findmkl.cmake [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  endif()
+else()
+   if(WIN32)
+      if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+         set(SEQ_LIB "mkl_sequential_dll.lib")
+         set(INT_LIB "mkl_intel_lp64_dll.lib")
+         set(COR_LIB "mkl_core_dll.lib")
+      else()
+         set(SEQ_LIB "mkl_sequential_dll.lib")
+	     set(INT_LIB "mkl_intel_c_dll.lib")
```
Comment (NeroBurner) [inline cmake/Findmkl.cmake]: mixed tab and spaces, please fix

File: cmake/Findmkl.cmake [hunk 6]
```diff
@@ -0,0 +1,104 @@
+if (MKL_INCLUDE_DIRS AND MKL_LIBRARIES AND MKL_INTERFACE_LIBRARY AND
+    MKL_SEQUENTIAL_LAYER_LIBRARY AND MKL_CORE_LIBRARY)
+  set (MKL_FIND_QUIETLY TRUE)
+endif()
+if(NOT MKL_SHARED_LIBS)
+  if(WIN32)
+      if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+          set(SEQ_LIB "mkl_sequential.lib")
+          set(INT_LIB  "mkl_intel_lp64.lib")
```
Comment (NeroBurner) [inline cmake/Findmkl.cmake]: one additional space, messes with alignment

File: cmake/Findmkl.cmake [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+          set(COR_LIB "mkl_core.lib")
+      else()
+          set(SEQ_LIB "mkl_sequential.lib")
+          set(INT_LIB  "mkl_intel_c.lib")
```
Comment (NeroBurner) [inline cmake/Findmkl.cmake]: one additional space, messes with alignment

File: cmake/Findmkl.cmake [hunk 8]
```diff
@@ -0,0 +1,104 @@
+if (MKL_INCLUDE_DIRS AND MKL_LIBRARIES AND MKL_INTERFACE_LIBRARY AND
```
Comment (NeroBurner) [inline cmake/Findmkl.cmake]: add a header (block of comments) which explains the variables used by this Findmkl module and which variables are set

File: cmake/Findmkl.cmake [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+                       $ENV{MKLROOT}/lib/ia32
+                       $ENV{INTEL}/mkl/lib/ia32
+                 NO_DEFAULT_PATH)
+
+     find_library(MKL_INTERFACE_LIBRARY
+                 NAMES ${INT_LIB}
+                 PATHS $ENV{MKLROOT}/lib
+                       $ENV{MKLROOT}/lib/ia32
+                       $ENV{INTEL}/mkl/lib/ia32
+                 NO_DEFAULT_PATH)
```
Comment (NeroBurner) [inline cmake/Findmkl.cmake]: no system wide installed packages are searched. For example unter Linux the default paths are `/usr/lib64/` or `/usr/lib`. You could add a final find_library command at the end. You don't even have to put the find_library call into an if-else block

> If the library is found the result is stored in the variable and the search will not be repeated unless the variable is cleared

File: cmake/Findmkl.cmake [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+else()
+
+  set(MKL_INCLUDE_DIRS "")
+  set(MKL_LIBRARIES "")
+  set(MKL_SEQUENTIAL_LAYER_LIBRARY "")
+  set(MKL_CORE_LIBRARY "")
+
+endif()
+
+INCLUDE(FindPackageHandleStandardArgs)
```
Comment (NeroBurner) [inline cmake/Findmkl.cmake]: all other keywords used are lower case, please make `INCLUDE`, `FIND_PACKAGE_HANDLE_STANDARD_ARGS` and `MARK_AS_ADVANCED` lower case too

**Comment by jlblancoc** [comment]
Closing, as I think it's superseded by #115 . Feel free to reopen if you think otherwise!

**Pull request closed without merging**